src/Microsoft.AspNet.Mvc.Core/Rendering/Html/TemplateRenderer.cs

```diff
@@ -167,36 +170,38 @@ private IEnumerable<string> GetViewNames()
                 }
```

Comment:
a couple of lines earlier (https://github.com/aspnet/Mvc/blob/file-tag-helper/src/Microsoft.AspNet.Mvc.Core/Rendering/Html/TemplateRenderer.cs#L152) to the end of the method are identical to the code in `InputTagHelper.GetInputTypeHints()`.  suggest adding a `public static GetTypeNames(Type type)` method in this class, avoiding the duplication.

admit that refactoring could have happened earlier but you're rearranging the code now...